rfcs/0192-min-capacity-ensures-workers-do-not-get-unnecessarily-killed.md

```diff
@@ -0,0 +1,58 @@
+# RFC 192 - `minCapacity` ensures workers do not get unnecessarily killed
+* Comments: [#192](https://github.com/taskcluster/taskcluster-rfcs/pull/192)
+* Proposed by: @JohanLorenzo
+
+# Summary
+
+`worker-manager` allows `minCapacity` to be set, ensuring a certain number of workers
+are available at any given time. Unlike what current happens now, these workers
+shouldn't be killed unless `minCapacity` is exceeded.
+
+## Motivation - why now?
+
+As far as I can remember, the current behavior has always existed. This year, the
+Engineering Effectiveness org is optimizing the cost of the Firefox CI instance.
+[Bug 1899511](https://bugzilla.mozilla.org/show_bug.cgi?id=1899511) made a change that
+actually uncovered the problem with the current behavior: workers gets killed after 2
+minuted and a new one gets spawned.
+
+
+# Details
+
+In the current implementation, workers are in charge of knowning when they have to shut
+down. Given the fact `docker-worker` is officially not supported anymore and we can't
+cut a new release and use it, let's change what config `worker-manager` gives to all
+workers, `docker-worker` included.
+
+## When `minCapacity` is exceeded
+
+In this case, nothing should change. `worker-manager` sends the same config to workers
+as it always did.
+
+## When `minCapacity` is not yet met
+
+Here, `worker-manager` should increase `afterIdleSeconds` to a much higher value (e.g.:
```

Comment:
"tweaking" idle seconds could be a good solution for existing workers that are unlikely to get upgraded soon.

There should be some upper limit of what worker-manager could set for such workers to balance `minCapacity` promise and avoiding additional costs from running them for too long

Reply:
To me, if we set `minCapacity` then we expect to have a number of workers at any time of the day. I would expect these workers to keep running. If some worker-types shouldn't be running for too long, I wonder if we should set `minCapacity` to 0.

Reply:
that's true! Indeed, if someone sets min value he probably knows what he's doing, no need to restrict

Reply:
Sorry term I think this isn't a problem. 

Long term something probabilistic could work: p(no) = 0.5 if exactly at minCapacity, p lower if higher than minCapacity